### PR TITLE
Labels to barplot and heatmap

### PR DESCRIPTION
**Pull request by katush**
Add label to barplot and heatmap if there is enough space - https://github.com/Caleydo/mothertable/issues/173

### CODE REVIEW

**Comment by thinkh** [comment]
One note: 
At the moment the text is overlaying the rect and thus prevents the selection of the row. In order to make the selection working, you need to add this lines of CSS:

```css
.phovea-heatmap { // for every vis type
  text {
    user-select: none;
    pointer-events: none;
  }
}
```

@katush Could you please add that to the default CSS?

**Comment by thinkh** [comment]
Please also add some padding to label so that it's not so close at the border.

vis-label-padding

**Comment by thinkh** [comment]
And please add the labels also to the `phovea-histogram` and `phovea-mosaic`. For the pie chart might be nice too, but might be to difficult.

**Comment by thinkh** [comment]
Additional request: Please take care that the font-size for a `phovea-list` and e.g., `phovea-heatmap` are equal for the same row height.

vis_match-font-sizes

File: src/heatmap/HeatMap1D.ts [hunk 1]
```diff
@@ -144,10 +146,32 @@ export default class HeatMap1D extends AVisInstance implements IVisInstance {
...
+
+      this.labels = $svg.append('g');
+      this.drawLabels();
       this.markReady();
     });
     return $svg;
   }
+
+  private drawLabels() {
+       const rowHeight = this.size[1] / this.data.dim[0];
```
Comment (thinkh) [inline src/heatmap/HeatMap1D.ts]: Incorrect line indent?

File: src/heatmap/HeatMap1D.ts [hunk 2]
```diff
@@ -144,10 +146,32 @@ export default class HeatMap1D extends AVisInstance implements IVisInstance {
...
+         'font-size' : rowHeight + 'px'
+       });
+      const t = <Promise<string|number[]>>this.data.data();
+      t.then((arr: any[]) => {
+        const $n = this.labels.selectAll('text').data(arr);
+        $n.enter().append('text');
+        const padding = 2;
+        $n.attr({
+          y: (d,i) => (i+1) * rowHeight - padding/2,
+          height: (d) => rowHeight- padding
```
Comment (thinkh) [inline src/heatmap/HeatMap1D.ts]: Space around minus

**Review by thinkh — CHANGES_REQUESTED** [review]
Implementation looks fine for me. Just minor formatting issues.

**Comment by katush** [comment]
@thinkh I am not sure how to do this, `phovea-list` is just text within divs. This is svg, the scaling works differently. Any ideas?

> Please take care that the font-size for a `phovea-list` and e.g., `phovea-heatmap` are equal for the same height.

**Comment by thinkh** [comment]
You can configure the initial scaling factor for the text (in the transform property) when you have both next to each other.

**Comment by katush** [comment]
image
 Should be done know. Though I consider moving the labels in histogram below it, same in filter view.

**Comment by thinkh** [comment]
Thanks. Now it looks good 👍

**Review by thinkh — APPROVED** [review]

**Pull request merged**